uaclient/files.py

```
@@ -350,6 +352,147 @@ def contract_id(self):
+    )
+    WRONG_FIPS_METAPACKAGE_ON_CLOUD = NoticeFileDetails(
+        label="wrong_fips_metapackage_on_cloud", order_id="25"
+    )
+    LIVEPATCH_LTS_REBOOT_REQUIRED = NoticeFileDetails(
+        label="lp_lts_reboot_required", order_id="30"
+    )
+    CONTRACT_REFRESH_WARNING = NoticeFileDetails(
+        label="contract_refresh_warning", order_id="40"
+    )
+    DAEMON_AUTO_ATTACH_LOCK_HELD = NoticeFileDetails(
+        label="daemon_auto_attach_lock_held", order_id="50"
+    )
+    DAEMON_AUTO_ATTACH_FAILED = NoticeFileDetails(
+        label="daemon_auto_attach_failed", order_id="51"
+    )
+    OPERATION_IN_PROGRESS = NoticeFileDetails(
+        label="operation_in_progress", order_id="60"
+    )
+
+
+class Notices:
+    def __init__(
+        self,
+        is_permanent: bool = True,
+        root_mode: bool = True,
+        override_dir: Optional[str] = None,
+    ):
+        if not override_dir:
+            self.directory = "/var/lib/ubuntu-advantage/notices/"
+            if not is_permanent:
+                self.directory = "/var/run/notices/"
+            self.override = False
+        else:
+            self.directory = override_dir
+            self.override = True
+        self.is_root = root_mode
+        self.is_permanent = is_permanent
+
+    def add(self, notice_label_id: NoticeFileDetails, description: str):
+        label = notice_label_id.label
+        file_order_id = notice_label_id.order_id
+        if self.is_root:
+            notice_files = self.get_all_notice_files()
+            labels_present = [file.split("-")[-1] for file in notice_files]
+            if label in labels_present:
+                return
+            filename = "{}-{}".format(file_order_id, label)
+            file = UAFile(filename, self.directory)
+            file.write(description)
+        else:
+            raise exceptions.NonRootUserError
+
+    def try_add(self, notice_label_id: NoticeFileDetails, description: str):
```

I think the normal `notices.add` should have the behavior of what is currently `try_add`. I can't think of a reason we'd want to raise an error when we use the wrong method - it should just log and do the right thing (not create the file if nonroot)